plaso/parsers/czip_plugins/interface.py

```diff
@@ -1,73 +1,64 @@
+  # REQUIRED_PATHS is a list of paths required by a plugin.
   # This is used to understand whether a plugin is suited for a given compound
-  # ZIP archive.
-  # This is expected to be overridden by actual plugins.
-  # Ex. frozenset(['[Content_Types].xml', '_rels/.rels', 'docProps/core.xml'])
-  REQUIRED_PATHS = frozenset(['any'])
+  # ZIP file.
+  # This must be overridden by actual plugins.
+  REQUIRED_PATHS = frozenset()
 
   NAME = 'czip'
 
-  def __init__(self):
-    super(CompoundZIPPlugin, self).__init__()
-    self._zip_file = None
-
-  def InspectArchive(self, parser_mediator):
-    """Inspects a compound ZIP archive looking for data.
+  @abc.abstractmethod
+  def InspectZipFile(self, parser_mediator, zip_file):
+    """Inspects a compound ZIP file and produces events.
 
     This is the main method that a compound ZIP plugin needs to implement.
 
-    Upon successful file check (always if you get down to this point) you will
-    find a file-like object corresponding to your archive in self._zip_file.
-    Feel free to implement the logic you prefer.
-
     Args:
       parser_mediator (ParserMediator): mediates interactions between parsers
           and other components, such as storage and dfvfs.
+      zip_file (zipfile.ZipFile): the zip file. It should not be closed in
+          this method, but will be closed by the parser logic in czip.py.
     """
 
   # pylint: disable=arguments-differ
-  def Process(self, parser_mediator, zip_file, name_list):
-    """Determine if this is the correct plugin; if so proceed with processing.
+  def Process(self, parser_mediator, zip_file, archive_members):
+    """Determines if this is the correct plugin; if so proceed with processing.
 
-    Process() checks if the current czip file being processed is a match for
-    a plugin by comparing REQUIRED_PATHS: if they match processing continues;
-    else raise WrongCompoundZIPPlugin.
+    This method checks if the zip file being contains the paths specified in
+    REQUIRED_PATHS. If all paths are present, the plugin logic processing
+    continues in InspectZipFile.
 
     Args:
       parser_mediator (ParserMediator): mediates interactions between parsers
           and other components, such as storage and dfvfs.
-      zip_file (zipfile.ZipFile): the archive file, already opened and
-          auto-closed.
-      name_list (list[str]): a convenience argument so we don't call
-          zip_file.namelist() once per plugin.
+      zip_file (zipfile.ZipFile): the zip file. It should not be closed in
+          this method, but will be closed by the parser logic in czip.py.
+      archive_members (list[str]): file paths in the archive.
 
     Raises:
       UnableToParseFile: when the file cannot be parsed.
+      ValueError: if a subclass has not specified REQUIRED_PATHS.
     """
-    if not set(name_list).issuperset(self.REQUIRED_PATHS):
-      raise errors.WrongCompoundZIPPlugin(self.NAME)
+    if not self.REQUIRED_PATHS:
```

Comment:
This may cause a problem in the future with localized czips, such as Google Takeout ones. Since filenames are localized, there is no way to tell if a given file is present inside the archive. With the previous implementation, one could leave this set empty on purpose and implement a custom logic [*], but now the whole thing needs to be reengineered.

Moreover, I've found a strange behaviour with both Python 2 and 3: in some cases [`ZipFile.namelist()`](https://docs.python.org/2/library/zipfile.html#zipfile.ZipFile.namelist) will list folder paths while in others won't; I've tested a couple of examples and I guess it depends on the library used to create the zip file, but I can further investigate.

In other words, given a context where there are some files (say _foo.txt_ and _bar.txt_) inside a directory (eg. _folder/_), `namelist()` may return either `['folder/', 'folder/foo.txt', 'folder/bar.txt']` or `['folder/foo.txt', 'folder/bar.txt']`.

[*]: I get it's not elegant, but it works

Reply:
As discussed, we'll need to look at this for future compound zips, but we'll go ahead with this approach for now.